terraform/cloudfront.tf

```
@@ -0,0 +1,100 @@
+
+locals {
+  s3_static_origin_id = "ashley-static-origin"
+  s3_media_origin_id = "ashley-media-origin"
+}
+
+# Create an origin access identity that will allow CloudFront to access S3
+# See bucket policies in s3.tf or documentation for more details:
+# https://www.terraform.io/docs/providers/aws/r/cloudfront_origin_access_identity.html
+resource "aws_cloudfront_origin_access_identity" "ashley_oai" {
+  comment = "fun-mooc origin for the ${terraform.workspace} environment"
+}
+
+resource "aws_cloudfront_distribution" "ashley_cloudfront_distribution" {
+  # Origin for static distribution
+  origin {
+    domain_name = lookup(var.app_domain, terraform.workspace)
+    origin_id   = local.s3_static_origin_id
+
+    custom_origin_config {
+      http_port = 80
+      https_port = 443
+      origin_protocol_policy = "https-only"
+      origin_ssl_protocols = ["SSLv3", "TLSv1", "TLSv1.1", "TLSv1.2"]
+    }
+  }
+  # Origin for the media S3 bucket
+  origin {
+    domain_name = aws_s3_bucket.ashley_media.bucket_domain_name
+    origin_id   = local.s3_media_origin_id
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.ashley_oai.cloudfront_access_identity_path
+    }
+  }
+
+  enabled         = true
+  is_ipv6_enabled = true
```

I suggest you to add the `wait_for_deployment` option to not have to wait in front of your terminal the end of the deployment.

```suggestion
  is_ipv6_enabled = true
  wait_for_deployment = false
```